pkg/zonemgr/zone_manager_test.go

```diff
@@ -0,0 +1,52 @@
+package zonemgr
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"os"
+
+	k8stypes "k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("Zone Manager functionality", func() {
+
+	const (
+		customDomain = "domain.com"
+		customNSIP   = "1.2.3.4"
+	)
+
+	var zoneMgr *ZoneManager
+
+	BeforeEach(func() {
+		os.Setenv("DOMAIN", customDomain)
+		os.Setenv("NAME_SERVER_IP", customNSIP)
+		zoneMgr = NewZoneManager()
+	})
+
+	AfterEach(func() {
+	})
+
+	Context("Initialization", func() {
+		It("should validate input data", func() {
+			Expect(zoneMgr.UpdateZone(k8stypes.NamespacedName{Name: "vm1"}, nil)).NotTo(Succeed())
+			Expect(zoneMgr.UpdateZone(k8stypes.NamespacedName{Namespace: "ns1"}, nil)).NotTo(Succeed())
+		})
+
+		It("should set custom data", func() {
+			Expect(zoneMgr.UpdateZone(k8stypes.NamespacedName{Namespace: "ns1", Name: "vm1"}, nil)).To(Succeed())
```

Comment:
`UpdateZone` is updating the `domain` and `nameServerIP`? I would expect `NewZoneManager` to do it.

Reply:
NewZoneManager indeed updates domain and nameServerIP. I use UpdateZone just to check that values are fetched and populated correctly